sdk/include/game-contract-sdk/game_base.hpp

```diff
@@ -266,32 +288,32 @@ class game: public eosio::contract {
             return;
         }
 
-        const auto req_id = get_req_id(memo);
+        const auto ses_id = get_ses_id(memo);
 
         eosio::check(quantity.symbol == core_symbol, "invalid token symbol");
 
-        const auto session_itr = sessions.find(req_id);
+        // if session doesn't exists create new session
+        if (sessions.find(ses_id) == sessions.end()) {
+            check_active_game();
 
-        if (session_itr == sessions.end()) {
-            eosio::check(platform::read::is_active_game(get_platform(), get_self_id()), "game is't listed in platform");
-
-            auto gl = global.get_or_default();
             sessions.emplace(get_self(), [&](auto& row){
-                row.req_id = req_id;
-                row.ses_seq = gl.session_seq++;
+                row.ses_id = ses_id;
+                row.ses_seq = global.session_seq++;
                 row.player = from;
                 row.deposit = quantity;
                 row.last_update = eosio::current_time_point();
                 row.last_max_win = zero_asset;
                 row.state = static_cast<uint8_t>(state::req_start);
             });
-            global.set(gl, get_self());
         }
-        else {
-            eosio::check(session_itr->player == from, "only player can deposit");
-            eosio::check(session_itr->state == static_cast<uint8_t>(state::req_deposit), "state should be 'req_deposit'");
-            eosio::check(!is_expired(req_id), "session expired");
-            sessions.modify(session_itr, get_self(), [&](auto& row){
+        else { /* else add extra deposit */
+            const auto& session = get_session(ses_id);
+
+            check_not_expired(session);
+            check_only_states(session, { state::req_deposit }, "state should be 'req_deposit'");
+            eosio::check(session.player == from, "only player can deposit");
+
+            sessions.modify(session, get_self(), [&](auto& row){
```

Comment:
I guess, it's no need to emit event here because after deposit we always transit to `req_action` state. In fact, the `req_deposit` state is for only internal use, and state change to `req_deposit` handles via `req_action` event with `need_deposit` flag.